agent/ebs/watcher.go

```diff
@@ -0,0 +1,177 @@
+// Copyright Amazon.com Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"). You may
+// not use this file except in compliance with the License. A copy of the
+// License is located at
+//
+//	http://aws.amazon.com/apache2.0/
+//
+// or in the "license" file accompanying this file. This file is distributed
+// on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+
+package ebs
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/aws/amazon-ecs-agent/agent/engine/dockerstate"
+	"github.com/aws/amazon-ecs-agent/agent/statechange"
+	apiebs "github.com/aws/amazon-ecs-agent/ecs-agent/api/resource"
+	log "github.com/cihub/seelog"
+)
+
+type EBSWatcher struct {
+	ctx        context.Context
+	cancel     context.CancelFunc
+	agentState dockerstate.TaskEngineState
+	// TODO: The ebsChangeEvent will be used to send over the state change event for EBS attachments once it's been found and mounted/resize/format.
+	ebsChangeEvent chan<- statechange.Event
+	// TODO: The dataClient will be used to save to agent's data client as well as start the ACK timer. This will be added once the data client functionality have been added
+	// dataClient     data.Client
```

Comment:
Q: sounds like a new boltDB bucket? Can you explain what will be persisted to boltDB? What's the key of the boltDB entry? How would we keep the task to the key map in state?

Reply:
Yep, that's correct. We'll need a new BoltDB bucket for EBS attachments. As for the key, we'll probably follow suit with how we're storing our ENI attachments into BoltDB where it parse the attachment ARN to obtain the key.  -> https://github.com/aws/amazon-ecs-agent/blob/master/agent/utils/utils.go#L231

Although since volume ID is also guaranteed to be unique, this could be a good option as well. This is still relative TBD but will have follow up PR to include these changes. Thanks!

Reply:
are we not also saving our EFS/FSx filesystems in state?